js/application.js

```diff
@@ -0,0 +1,298 @@
+$(document).ready(function(){
+
+  var winState=null;
+
+  var randomNumberGeneration = function(){
+  	var generated=false;
+  	while (generated==false){
+	    var colNum=Math.floor(Math.random()*3.99+1);
+	    var rowNum=Math.floor(Math.random()*3.99+1);
+	    gridCol="column"+colNum;
+	    gridRow="grid_"+rowNum;
+	    if($('#'+gridRow).children('#'+gridCol).text()=="0"){
+		    $('#'+gridRow).children('#'+gridCol).text("2");
+	  		generated=true;
+	    }
+  	}
+  }
+
+  var checkNoZeroes = function(){
+    var hasZeroes = false;
+    for(i=1; i<=4; i++){
+      var GridRow="grid_"+i;
+      for (j=1; j<=4; j++){
+        var GridCol="column"+j;
+        if ($('#'+GridRow).children('#'+GridCol).text()=='0'){
+          hasZeroes=true;
+        }
+      }
+    }
+    return !hasZeroes;
+  }
+
+  var reset = function(){
+    for(i=1; i<=4; i++){
+      var resetGridRow="grid_"+i;
+      for (j=1; j<=4; j++){
+        var resetGridCol="column"+j;
+        $('#'+resetGridRow).children('#'+resetGridCol).attr('myAttr','canMove');
+        $('#'+resetGridRow).children('#'+resetGridCol).attr('combine','canCombine');
+      }
+    }
+  }
+
+  var setColor = function(elementValue){
+    switch(elementValue.text()){
+      case "2":
+        elementValue.css('background-color','#EEE4DA');
+      case "4":
+        elementValue.css('background-color','#EDE0C8');
+      case "8":
+        elementValue.css('background-color','#F2B179');
+      case "16":
+        elementValue.css('background-color','#F59563');
+      case "32":
+        elementValue.css('background-color','#F67C5F');
+      case "64":
+        elementValue.css('background-color','#F65E3B');
+      case "128":
+        elementValue.css('background-color','#EDCF72');
+      case "256":
+        elementValue.css('background-color','#EDCC61');
+      case "512":
+        elementValue.css('background-color','#EDC850');
+      case "1024":
+        elementValue.css('background-color','#EDC53F');
+      case "2048":
+        elementValue.css('background-color','#EDC22E');
+      default
+        elementValue.css('background-color','#CCC0B3');
+    }
+  }
+	
+  randomNumberGeneration();
+  randomNumberGeneration();
+
+  var leftMove = function(){
+  	var hasMoved=false;
+  	for (i=1; i<=4; i++){
+  		gridRow="grid_"+i;
+  		$('#'+gridRow).children('#column1').attr('myAttr','cantMove');
+  		while(($('#'+gridRow).children('#column2').attr('myAttr')=='canMove')||($('#'+gridRow).children('#column3').attr('myAttr')=='canMove')||($('#'+gridRow).children('#column4').attr('myAttr')=='canMove')){
+  			for (j=2; j<=4; j++){
+  				gridCol="column"+j;
+  				gridColPrev="column"+(j-1);
+  				if($('#'+gridRow).children('#'+gridCol).text()!="0"){
+	  				if (($('#'+gridRow).children('#'+gridColPrev).text()=='0')) {
+	  					sum=Number($('#'+gridRow).children('#'+gridColPrev).text())+Number($('#'+gridRow).children('#'+gridCol).text());
+	  					$('#'+gridRow).children('#'+gridColPrev).text(sum);
```

Comment:
You should store `$('#'+gridRow).children('#'+gridColPrev)` into a descriptive variable since you are using it repeatedly